timesketch/tsctl.py

```diff
@@ -1373,3 +1416,574 @@ def list_config():
+
+                output_jsonl = io.StringIO()
+                count = 0
+                for row in reader:
+                    processed_row = {}
+                    for key, value in row.items():
+                        try:
+                            processed_row[key] = int(value)
+                            continue
+                        except (ValueError, TypeError):
+                            pass
+                        try:
+                            processed_row[key] = float(value)
+                            continue
+                        except (ValueError, TypeError):
+                            pass
+                        if isinstance(value, str):
+                            if value.lower() == "true":
+                                processed_row[key] = True
+                                continue
+                            if value.lower() == "false":
+                                processed_row[key] = False
+                                continue
+                        processed_row[key] = value
+
+                    output_jsonl.write(json.dumps(processed_row) + "\n")
+                    count += 1
+
+                event_data_bytes = output_jsonl.getvalue().encode("utf-8")
+                print(f"    Successfully converted {count} CSV rows to JSONL.")
+
+            except Exception as conversion_error:  # pylint: disable=broad-except
+                print(f"  ERROR converting CSV to JSONL: {conversion_error}")
+                print(traceback.format_exc())
+                raise  # Re-raise to stop execution
+
+    return event_data_bytes
+
+
+# Helper function to create the zip archive
+# TODO(jaegeral): https://github.com/google/timesketch/issues/3415
+def _create_export_archive(
+    filename: str, metadata: dict, event_data_bytes: bytes, event_filename: str
+):
+    """Creates the zip archive with metadata and event data."""
+    print("Creating zip archive...")
+    try:
+        with zipfile.ZipFile(filename, "w", zipfile.ZIP_DEFLATED) as zipf:
+            metadata_bytes = json.dumps(metadata, indent=2, ensure_ascii=False).encode(
+                "utf-8"
+            )
+            zipf.writestr(DEFAULT_EXPORT_METADATA_FILENAME, metadata_bytes)
+
+            if event_data_bytes:
+                zipf.writestr(event_filename, event_data_bytes)
+            else:
+                print("  WARNING: No event data was generated to include in the zip.")
+
+        print(f"Sketch exported successfully to {filename}")
+
+    except Exception as zip_error:  # pylint: disable=broad-except
+        print(f"ERROR creating zip file: {zip_error}")
+        print(traceback.format_exc())
+
+
+# --- Main CLI command using the helper functions ---
+@cli.command(name="export-sketch")
+@click.argument("sketch_id", type=int)
+@click.option(
+    "--output-format",
+    type=click.Choice(["csv", "jsonl"], case_sensitive=False),
+    default="csv",
```

Comment:
Given opensearch returns json, shouldn't we go with `jsonl` as default so we save the converting step?

Reply:
see above